HPXMLtoOpenStudio/resources/airflow.rb

```diff
@@ -1636,41 +1561,105 @@ def self.apply_infiltration_and_ventilation_fans(model, weather, vent_mech, vent
+      infil_program.addLine('If Qfan > 0')
+      # Balanced system if the total supply airflow and total exhaust airflow are within 10% of their average.
+      infil_program.addLine('  Set Qavg = ((Qexhaust + Qsupply) / 2.0)')
+      infil_program.addLine('  If ((@Abs (Qexhaust - Qavg)) / Qavg) <= 0.1') # Only need to check Qexhaust, Qsupply will give same result
+      infil_program.addLine('    Set phi = 1')
+      infil_program.addLine('  Else')
+      infil_program.addLine('    Set phi = (Qinf / (Qinf + Qfan))')
+      infil_program.addLine('  EndIf')
+      infil_program.addLine('  Set Qinf_adj = phi * Qinf')
+      infil_program.addLine('Else')
+      infil_program.addLine('  Set Qinf_adj = Qinf')
+      infil_program.addLine('EndIf')
+    else
+      infil_program.addLine('Set Qu = (@Abs (Qexhaust - Qsupply))') # Unbalanced flow
+      infil_program.addLine('Set Qb = Qfan - Qu') # Balanced flow
+      infil_program.addLine('Set Qtot = (((Qu^2) + (Qinf^2)) ^ 0.5) + Qb')
+      infil_program.addLine('Set Qinf_adj = Qtot - Qu - Qb')
+    end
 
     # Assign total airflow to different component loads
-    infil_program.addLine("Set #{imbal_mechvent_flow_actuator.name} = (@Abs ((Qout - QductsOut) - (Qin - QductsIn)))")
-    infil_program.addLine("Set #{imbal_ducts_flow_actuator.name} = (@Abs (QductsOut - QductsIn))")
-
-    # Assign the remainder to infiltration:
-    infil_program.addLine("Set #{infil_flow_actuator.name} = Q_tot_flow - #{imbal_mechvent_flow_actuator.name} - #{imbal_ducts_flow_actuator.name}")
-    infil_program.addLine("If #{infil_flow_actuator.name} < 0")
-    infil_program.addLine("  Set #{infil_flow_actuator.name} = 0")
-    infil_program.addLine('EndIf')
+    if (not vent_mech.nil?) && [HPXML::MechVentTypeERV, HPXML::MechVentTypeHRV].include?(vent_mech.fan_type)
+      infil_program.addLine("Set #{mechvent_flow_actuator.name} = Qfan - QWHV") # QWHV load captured by ERV/HRV program below
+    else
+      infil_program.addLine("Set #{mechvent_flow_actuator.name} = Qfan")
+    end
+    infil_program.addLine("Set #{infil_flow_actuator.name} = Qinf_adj")
+
+    if (not vent_mech.nil?) && [HPXML::MechVentTypeERV, HPXML::MechVentTypeHRV].include?(vent_mech.fan_type) && (vent_mech_cfm > 0)
+      # ERV/HRV EMS load model
+      # E+ ERV model uses standard density for MFR calculation, causing discrepancy with other system types, so
+      # ERV is modeled within EMS infiltration program instead.
+
+      # Sensors for ERV/HRV
+      win_sensor = OpenStudio::Model::EnergyManagementSystemSensor.new(model, 'Zone Air Humidity Ratio')
+      win_sensor.setName("#{Constants.ObjectNameAirflow} win s")
+      win_sensor.setKeyName(@living_zone.name.to_s)
+
+      # Actuators for ERV/HRV
+      sens_name = "#{Constants.ObjectNameERVHRV} sensible load"
+      erv_sens_load_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, sens_name.gsub(' ', '_'))
```

Comment:
The global variable is not used anywhere, it's created because the method of `create_sens_lat_load_actuator_and_equipment` below was, at one point, combined with duct actuators along with their global variable creations. So feel free to remove these two lines if you want.

Reply:
Cool, thanks for pointing that out. I'll double-check it and remove it if you're right. I didn't actually touch your code at all, just moved where it gets called.